docs/error-codes.md

```diff
@@ -266,4 +472,154 @@ the error code. For example:
 
 #### `IL2049`: Unrecognized internal attribute 'attribute'
 
-- The internal attribute name 'attribute' being used in the xml is not supported by the linker, check the spelling and the supported internal attributes.
\ No newline at end of file
+- The internal attribute name 'attribute' being used in the xml is not supported by the linker, check the spelling and the supported internal attributes.
+
+#### `IL2050` Trim analysis: Making a generic type instantiation from 'type' which has 'DynamicallyAccessedMembersAttribute' on some of its generic parameters. ILLink currently doesn't analyze type values for generic parameters when making a generic type instantiation via 'System.Type.MakeGenericType'
+
+- Analysis of type parameters for `System.Type.MakeGenericType` is not yet implemented. If the open generic type `type` has `DynamicallyAccessedMembersAttribute` on any of its generic parameters, ILLink currently can't validate that the requirements are fulfilled by the calling method.  
+
+  ``` C#
+  class Lazy<[DynamicallyAccessedMembers(DynamicallyAccessedMemberType.PublicParameterlessConstructor)] T> 
+  {
+      // ...
+  }
+  
+  void TestMethod()
+  {
+      // IL2050 Trim analysis: Making a generic type instantiation from 'Lazy<T>' which has 
+      // 'DynamicallyAccessedMembersAttribute' on some of its generic parameters. ILLink currently doesn't analyze 
+      // type values for generic parameters when making a generic type instantiation via 'System.Type.MakeGenericType'
+      typeof(Lazy<>).MakeGenericType(new Type[] { typeof(TestType) });
+  }
+  ```
+
+  #### `IL2051` Trim analysis: Unrecognized type value used to called the method 'MakeGenericType'. It's not possible to guarantee the availability of requirements of such type.
```

Comment:
I don't understand the warning message :-(

Reply:
As per the code sample - calling `MakeGenericType` on a `Type` instance we can't statically analyze - if that type has annotations on its generic parameters, we would be required to validate those - but can't since we don't know the type.
If anybody has a better idea about the message, I'm more than happy to change the text.

Reply:
How is that different from IL2052? Do we need to expose the limitation over warning codes?

Does it make sense to tweak IL2052 to cover all such cases?

```
#### `IL2052` Method `MakeGenericType(...)` is called on instance which cannot be determined.
```

Reply:
That message is also not correct - but what about this:
```
IL2051 Trim analysis: Call to `System.Reflection.MethodInfo.MakeGenericType` can not be statically analyzed. It's not possible to guarantee the availability of requirements of the generic type.
```

Basically the same as IL2056 but for types. And to avoid future question - I do want to keep the `MakeGenericType` and `MakeGenericMethod` warnings separate. If nothing else, to give us freedom of improving the warning message if the current one is not good enough (and sharing it between both cases would make it much harder to come up with a message)

Reply:
Yep, this is better